### PR TITLE
Global improvements

### PR DESCRIPTION
**Pull request by CPatchane**
~~__⚠️ Must be merged only after #31 ⚠️__~~

This PR bring global improvements to the application.

* Remove photos selection when added with success to an album
* Remove photos selection when entering into an album
* Add fetching error view to the photos list view
* Better error and empty messages
* Improve greatly tests

### CODE REVIEW

File: src/lib/helpers.js [hunk 1]
```diff
@@ -0,0 +1,21 @@
+export const mustShowSelectionBar = state => state.ui.showSelectionBar || state.ui.selected.length !== 0
```
Comment (y-lohse) [inline src/lib/helpers.js]: The only part of the PR I'm not too sure about is this one.

I agree that these functions are not reducers; however, they do compute things based on the `state`, so I think they qualify as selectors. As such, they are closer to reducers than to independent helper functions.
Files has similar code and it's still with the reducers. So yeah, not too sure this is a good move.

Can we get a third opinion? @goldoraf or @gregorylegarec maybe?

Comment (goldoraf) [inline src/lib/helpers.js]: Yep, they are selectors, and as such, should be colocated with reducers.

File: src/actions/albums.js [hunk 2]
```diff
@@ -45,6 +48,30 @@ export const fetchAlbums = (mangoIndex) => {
...
+      let fetchedPhotos = []
+      for (let index in photosIds) {
+        const photo = await cozy.client.files.statById(photosIds[index])
+        fetchedPhotos.push(photo)
+      }
+      album.photos = fetchedPhotos
+      dispatch({type: FETCH_CURRENT_ALBUM_PHOTOS_SUCCESS, album})
+      return fetchedPhotos
+    } catch (fetchError) {
+      throw new Error(fetchError.response
```
Comment (goldoraf) [inline src/actions/albums.js]: These 3 lines of error handling are repeated several times in this file. Maybe a small refactoring here?

File: src/lib/helpers.js (hunk 1 above)
Comment (gregorylegarec) [inline src/lib/helpers.js]: My opinion is : if it compute something from state, it's a selector. Am I right @goldoraf ?

File: src/containers/AlbumPhotos.jsx [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+export class AlbumPhotos extends Component {
+  constructor (props) {
+    super(props)
+    this.state = {
+      photosAreDirty: false
+    }
+  }
+
+  componentWillUpdate (nextProps, nextState) {
+    this.state.photosAreDirty = nextProps.photos &&
```
Comment (goldoraf) [inline src/containers/AlbumPhotos.jsx]: You must not alter the state directly, but use `setState` instead. BUT I think it won't really work here because setting state in `componentWillUpdate` will trigger another rerender. You may want to find another way to do this.

Comment (CPatchane) [inline src/containers/AlbumPhotos.jsx]: As discussed, this part will be changed in another pull request. Thanks 👍

File: src/actions/albums.js (hunk 2 above)
Comment (CPatchane) [inline src/actions/albums.js]: You're right. I fixed it 👍
